lucille-core/src/main/java/com/kmwllc/lucille/connector/jdbc/DatabaseConnector.java

```diff
@@ -253,18 +252,20 @@ private void iterateOtherSQL(ResultSet rs2, String[] columns2, Document doc, Int
       // um... why is this getting called?  if it is?
       return;
     }
-
     do {
       // Convert to do-while I think we can avoid the rs2.previous() call.
-      // TODO: support non INT primary key
-      int otherJoinId = rs2.getInt(joinField);
-      if (otherJoinId < joinId) {
+      Object otherJoinId = rs2.getObject(joinField);
+
+      if (!(otherJoinId instanceof Comparable) || !(joinId instanceof Comparable) || !otherJoinId.getClass().equals(joinId.getClass())) {

```

Comment:
is there a way to do this check only once -- somewhere outside of the do/while?

Reply:
yes. Resolved in [cfbc0e0](https://github.com/kmwtechnology/lucille/pull/184/commits/cfbc0e07bb2e737c6c86c682d3dbb9e8d9241c87)